waku/v2/node/peer_manager.nim

```diff
@@ -0,0 +1,67 @@
+{.push raises: [Defect, Exception].}
+
+import
+  chronos, chronicles,
+  libp2p/standard_setup,
+  libp2p/peerstore
+
+type
+  PeerManager* = ref object of RootObj
+    switch*: Switch
+    peerStore*: PeerStore
+
+proc new*(T: type PeerManager, switch: Switch): PeerManager =
+  T(switch: switch,
+    peerStore: PeerStore.new())
+
+####################
+# Dialer interface #
+####################
+
+proc dialPeer*(pm: PeerManager, peerInfo: PeerInfo, proto: string): Future[Connection] {.async.} =
```

Comment:
What happens if this call fails?

Reply:
Currently still a very ungraceful unhandled `DialFailedError`. The plan is to timeout the `dial` future on the `switch`, log these failed/timed-out connections and keep track of such peers (e.g. changing their `connectedness` state to `CannotConnect` for a period of time). I'm also considering adding at least one redial attempt per failure, if there is a sensible way of doing this.